src/Default/default.ts

```diff
@@ -36,13 +40,79 @@ export class DefaultContainerWindow implements ContainerWindow {
     }
 }
 
-declare var Notification: any;
+/**
+ * @augments MessageBus
+ */
+export class DefaultMessageBus implements MessageBus {
+    private container: DefaultContainer;
+    private static readonly messageSource: string = "desktopJS";
+
+    constructor(container: DefaultContainer) {
+        this.container = container;
+    }
+
+    subscribe<T>(topic: string, listener: (event: any, message: T) => void, options?: MessageBusOptions): Promise<MessageBusSubscription> {
+        return new Promise<MessageBusSubscription>((resolve, reject) => {
+            const subscription: MessageBusSubscription = new MessageBusSubscription(topic, (event: any) => {
+                // Only allow same origin
+                if (event.origin !== this.container.globalWindow.location.origin) {
+                    return;
+                }
+
+                // Make sure topic received matches the one that was subscribed
+                const { source, "topic": receivedTopic, message } = JSON.parse(event.data);
+
+                if (source === DefaultMessageBus.messageSource && topic === receivedTopic) {
+                    listener({ topic: topic }, message);
```

Comment:
To avoid any confusion, the line you see in default.ts is related to CORS for origin.  It is ignoring messages sent from other site origins than the subscriber to prevent any type of attack.  Send message works cross origin so you have to manually check.